packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -454,31 +457,46 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 			0x385 /* For clarity and paranoid defense against adding future attachment states */,
 		);
 
+		if (signal?.aborted) {
+			throw(Error("aborted before uploading"));//signal as any).reason));
+		}
+
 		// Create a local ID for the blob. After uploading it to storage and before returning it, a local ID to
 		// storage ID mapping is created.
 		const localId = uuid();
 		const pendingEntry: PendingBlob = {
 			blob,
 			status: PendingBlobStatus.OnlinePendingUpload,
 			handleP: new Deferred(),
-			uploadP: this.uploadBlob(localId, blob),
+			uploadP: this.uploadBlob(localId, blob, signal),
 			attached: false,
 			acked: false,
 		};
 		this.pendingBlobs.set(localId, pendingEntry);
 
+		if(signal){
+			signal.addEventListener('abort', () => {
+				this.deletePendingBlob(localId);
+				pendingEntry.handleP.reject(Error("aborted while uploading"));
+			}, {once: true});
+		}
+
 		return pendingEntry.handleP.promise;
 	}
 
-	private async uploadBlob(localId: string, blob: ArrayBufferLike): Promise<ICreateBlobResponse> {
+	private async uploadBlob(
+		localId: string,
+		blob: ArrayBufferLike,
+		signal?: AbortSignal,
```

Comment:
rather than passing it everywhere can we just store it on the pending entry

Reply:
The pending entry is removed in the abort event, so in case of aborting while uploading, once it finishes there won't be an entry to check the signal.

Reply:
i don't know if that is the correct pattern. i think removing the pending entry on abort could make deal with the op flows harder, or less safe, as you will want to know if exists, and was aborted, rather than did not exist